### ISSUE TITLE
Erratic behavior when blind target is beyond RA limit

### ISSUE THREAD

**Issue by bgottula** [bgottula/track#73]
On 6 March I started the blind_track program while the target satellite was still well below the horizon. The mount tracked until it hit the RA limit but then started making some strange sounds and little flinches that didn’t seem healthy. I stopped and restarted it a couple of times but this behavior persisted so I stopped it until the sat was above the horizon before continuing. Not sure what the root cause is but this needs to be fixed.

**Comment by bgottula** [comment]
Confirmed that this is not handled well by using hybrid_track to slew to star Capella on meridian side east, causing the RA axis to hit the limit. The mount did stop as expected and there was no obviously erratic motion but the period of the control loop was very slow and erratic. On at least one occasion a loop filter did not update for a full 32 seconds! Based on the values here it seems that the second of the two slew commands is taking way too long to complete. This is still odd though because the Gemini2BackendUDP object is constructed with a default timeout of 0.25 seconds. I thought I would at least see some kind of message when that timeout expired. Hmmm.

```
time: 1525236990.7592082 slew rates: {'ra': 0.0, 'dec': 0.1636687204544854}
Warning: loop filter update period was 0.10100269317626953 s, limit is 0.1 s.
Warning: loop filter update period was 1.5320067405700684 s, limit is 0.1 s.
time: 1525236992.291183 slew rates: {'ra': 0.0, 'dec': 0.1636687204544854}
Warning: loop filter update period was 0.10132861137390137 s, limit is 0.1 s.
Warning: loop filter update period was 0.7686631679534912 s, limit is 0.1 s.
time: 1525236993.0598426 slew rates: {'ra': 0.0, 'dec': 0.1636687204544854}
Warning: loop filter update period was 0.10317778587341309 s, limit is 0.1 s.
Warning: loop filter update period was 6.273949861526489 s, limit is 0.1 s.
time: 1525236999.3337805 slew rates: {'ra': 0.0, 'dec': 0.1636687204544854}
Warning: loop filter update period was 0.10088419914245605 s, limit is 0.1 s.
Warning: loop filter update period was 3.13427472114563 s, limit is 0.1 s.
time: 1525237002.468068 slew rates: {'ra': 0.0, 'dec': 0.1636687204544854}
Warning: loop filter update period was 0.10127019882202148 s, limit is 0.1 s.
Warning: loop filter update period was 32.55070090293884 s, limit is 0.1 s.
time: 1525237035.018772 slew rates: {'ra': 0.0, 'dec': 0.1636687204544854}
Warning: loop filter update period was 0.10028481483459473 s, limit is 0.1 s.
Warning: loop filter update period was 0.8265492916107178 s, limit is 0.1 s.
time: 1525237035.8453155 slew rates: {'ra': 0.0, 'dec': 0.1636687204544854}
Warning: loop filter update period was 0.10048913955688477 s, limit is 0.1 s.
Warning: loop filter update period was 3.1360106468200684 s, limit is 0.1 s.
time: 1525237038.9797485 slew rates: {'ra': 0.0, 'dec': 0.1636687204544854}
Warning: loop filter update period was 0.10097622871398926 s, limit is 0.1 s.
Warning: loop filter update period was 3.134542465209961 s, limit is 0.1 s.
```

**Comment by bgottula** [comment]
In addition to the bad control loop behavior the hand controller also becomes unresponsive. To recover I have to unplug and re-plug it into Gemini 2.

A theory I have is that Gemini 2 behaves badly when its internal limit is reached. If we set the Gemini 2 limit to something beyond the limit set in software perhaps the problem will go away.

**Comment by bgottula** [comment]
I tried setting the Gemini2 axis limits to 150 degrees on both sides of the meridian but this didn't seem to help.

I think I found a problem with how `AxisLimitException` is handled in `Tracker.run()` which may explain the problem. When this exception is thrown by first axis in the for loop, control goes to the exception handler. This prevents the second axis from getting a chance to run. This explains why the second loop filter often doesn't update for seconds at a time. Reworking how these exceptions are handled should improve the situation.
